File: mvp/src/stores/chatStore.ts
import { create } from 'zustand'
import { persist } from 'zustand/middleware'

export interface ChatMessage {
  id: string
  role: 'user' | 'assistant'
  content: string
}

interface Config {
  provider: 'mock' | 'ollama'
  ollamaUrl: string
  model: string
  voiceRate: number
  voicePitch: number
  theme: 'light' | 'dark'
  avatar: string
}

interface ChatStore {
  messages: ChatMessage[]
  config: Config
  isAlarmSet: boolean
  alarmRemaining: number | null
  isListening: boolean
  transcript: string
  addMessage: (msg: Omit<ChatMessage, 'id'>) => void
  setMessages: (messages: ChatMessage[]) => void
  updateConfig: (config: Partial<Config>) => void
  setIsListening: (v: boolean) => void
  setTranscript: (v: string) => void
  setAlarmRemaining: (v: number | null) => void
  setIsAlarmSet: (v: boolean) => void
  clearMessages: () => void
}

export const useChatStore = create<ChatStore>()(
  persist(
    (set) => ({
      messages: [
        {
          id: 'welcome',
          role: 'assistant',
          content: '你好！我是 Voicer，有什么可以帮你的吗？',
        },
      ],
      config: {
        provider: 'mock',
        ollamaUrl: 'http://localhost:11434',
        model: 'llama3.2',
        voiceRate: 1,
        voicePitch: 1,
        theme: 'light',
        avatar: '',
      },
      isListening: false,
      transcript: '',
      isAlarmSet: false,
      alarmRemaining: null,
      addMessage: (msg) =>
        set((s) => ({
          messages: [...s.messages, { ...msg, id: crypto.randomUUID() }],
        })),
      setMessages: (messages) => set({ messages }),
      updateConfig: (cfg) =>
        set((s) => ({ config: { ...s.config, ...cfg } })),
      setIsListening: (v) => set({ isListening: v }),
      setTranscript: (v) => set({ transcript: v }),
      setAlarmRemaining: (v) => set({ alarmRemaining: v }),
      setIsAlarmSet: (v) => set({ isAlarmSet: v }),
      clearMessages: () => set({ messages: [] }),
    }),
    {
      name: 'voicer-chat',
      partialize: (state) => ({
        messages: state.messages,
        config: {
          provider: state.config.provider,
          ollamaUrl: state.config.ollamaUrl,
          model: state.config.model,
          voiceRate: state.config.voiceRate,
          voicePitch: state.config.voicePitch,
          theme: state.config.theme,
          avatar: state.config.avatar,
        },
      }),
    }
  )
)

File: ios/Sources/Stores/ChatStore.swift
import Foundation
import SwiftUI

@Observable
final class ChatStore {
    var messages: [Message] = [
        Message(role: .assistant, content: "你好！我是 Voicer，有什么可以帮你的吗？")
    ]
    var config: AppConfig = .defaults
    var isListening: Bool = false
    var transcript: String = ""
    var isAlarmSet: Bool = false
    var alarmRemaining: Int? = nil

    private let userDefaultsKey = "voicer-chat-store"

    init() {
        load()
    }

    func addMessage(_ message: Message) {
        messages.append(message)
        save()
    }

    func sendUserMessage(content: String) {
        let msg = Message(role: .user, content: content)
        addMessage(msg)
    }

    func updateConfig(_ config: AppConfig) {
        self.config = config
        save()
    }

    func clearMessages() {
        messages.removeAll()
        messages.append(Message(role: .assistant, content: "你好！我是 Voicer，有什么可以帮你的吗？"))
        save()
    }

    private func save() {
        if let data = try? JSONEncoder().encode(config) {
            UserDefaults.standard.set(data, forKey: userDefaultsKey + "-config")
        }
        if let data = try? JSONEncoder().encode(messages) {
            UserDefaults.standard.set(data, forKey: userDefaultsKey + "-messages")
        }
    }

    private func load() {
        if let data = UserDefaults.standard.data(forKey: userDefaultsKey + "-config"),
           let cfg = try? JSONDecoder().decode(AppConfig.self, from: data) {
            config = cfg
        }
        if let data = UserDefaults.standard.data(forKey: userDefaultsKey + "-messages"),
           let msgs = try? JSONDecoder().decode([Message].self, from: data) {
            if !msgs.isEmpty {
                messages = msgs
            }
        }
    }
}
